lib/env/env.c

```diff
@@ -171,3 +172,56 @@ void spdk_delay_us(unsigned int us)
 {
 	return rte_delay_us(us);
 }
+
+#define BDEV_SYS_CPU_DIR "/sys/devices/system/cpu/cpu%u"
+#define BDEV_CORE_ID_FILE "topology/core_id"
+#define BDEV_CPU_PATH_MAX 256
+
+/* Check if a cpu is present by the presence of the cpu information for it */
+static int blockdev_cpu_detected(unsigned lcore_id)
+{
+	char path[BDEV_CPU_PATH_MAX];
+	int len = snprintf(path, sizeof(path), BDEV_SYS_CPU_DIR
+		"/"BDEV_CORE_ID_FILE, lcore_id);
+	if (len <= 0 || (unsigned)len >= sizeof(path))
+		return 0;
+	if (access(path, F_OK) != 0)
+		return 0;
+
+	return 1;
+}
+
+int spdk_use_all_cpu_cores_for_app(app_callback cb, void *args)
```

Comment:
I don't really like this name, but I am having a hard time coming up with a better one.

Maybe spdk_call_unaffinitized()?